web/reset/index.php

```diff
@@ -52,24 +60,34 @@
             $data = json_decode(implode('', $output), true);
             $rkey = $data[$user]['RKEY'];
             if (hash_equals($rkey, $_POST['code'])) {
-                $v_password = tempnam("/tmp","vst");
-                $fp = fopen($v_password, "w");
-                fwrite($fp, $_POST['password']."\n");
-                fclose($fp);
-                $cmd="/usr/bin/sudo /usr/local/hestia/bin/v-change-user-password";
-                exec ($cmd." ".$v_user." ".$v_password, $output, $return_var);
-                unlink($v_password);
-                if ( $return_var > 0 ) {
-                    $ERROR = "<a class=\"error\">".__('An internal error occurred')."</a>";
-                } else {
-                    $_SESSION['user'] = $_POST['user'];
-                    header("Location: /");
-                    exit;
+                unset($output);
+                exec('/usr/bin/sudo /usr/local/hestia/bin/v-get-user-value '.$v_user.' RKEYEXP', $output,$return_var);
```

Comment:
don't use single quotes around $v_user, it is escaped earlier with escapeshellarg

Reply:
Oke changed it..